Automattic/jetpack
# VideoPress: Add wpcom support to Ajax class

We are looking to share the same `class-ajax.php` code with Jetpack and wp.com. During testing I discovered that on wpcom it was trying to use the `shadow_blog_id` which isn't available on wp.com simple sites.

## Proposed changes:

* This PR adds a helper function to get the appropriate blog id depending on if we are on Jetpack or wp.com.

### Other information:


## Jetpack product discussion


pxWta-1P4-p2

## Does this pull request change what data or activity we track or use?


No

## Testing instructions:






* From a Jetpack site, ensure VideoPress is enabled and upload a video in the block editor. It should succeed.
* Apply patch to wp.com (`bin/jetpackddownloader test jetpack update/videopress-ajax-wpcom-support`), apply VideoPress HQ and attempt to add a new video, you should see the edit fields which means the ajax call succeeded.
